### DISCUSSION TITLE
Set `SDL_VIDEODRIVER=wayland` on Wayland by default

### DISCUSSION THREAD

**Discussion by easyteacher** [ppy/osu#22651] [Ideas] [answered]
When running osu! in a Wayland session, without `SDL_VIDEODRIVER=wayland` the game is basically unplayable (high latency). Request to add `SDL_VIDEODRIVER=wayland` in AppRun to make the game playable on Wayland.

**Comment by bdach** [comment]
If you search around on this repository even, we've had users report issues due to `SDL_VIDEODRIVER=wayland` **being set** *and* due to it **not** being set. Wayland was the default video driver for a bit when SDL switched to use it by default upstream, but then I believe reverted that when it came out that Wayland support is still not up to scratch. I would much rather defer to upstream on the choice of default. I have no data that indicates that setting `SDL_VIDEODRIVER=wayland` by default is going to be an universal improvement. We're going to be seeing wayland issues anyway because seemingly every linux installation is a special snowflake and breaks in a slightly different way and none of the issues can be even reproduced and debugged without having that specific environment.

**Comment by orowith2os** [comment]
Wayland appears to be the default again in upstream SDL2, so focusing more on that would probably be useful. 

The improvement is a simpler API underneath everything; Xorg is an unmaintainable hell, and Xwayland is no better. Plus Wayland has proper fractional scaling, something I'm sure gamers would like (no downsampling). 

Better to deal with Wayland issues where you can than put it off and end up making it harder down the line.

**Reply by bdach** [reply, marked as answer]
If upstream has switched back to wayland by default then eventually so will we. No more discussion is necessary.

Any serious windowing issues are upstream's to deal with and not ours anyway.

**Comment by kawaii-ghost** [comment]
How bout also set `SDL_AUDIODRIVER=pipewire` ?

**Reply by bdach** [reply]
We don't use sdl for audio.

**Reply by djsiropchik** [reply]
> We don't use sdl for audio.

okay. So what do you use guys? Cause all the time just in one moment sound is gone and I need to change audio device to other and back sometimes in the game, sometimes in default settings of the system. It's annoying. Fedora 43, pipewire, Plasma 6.5

**Reply by sineplusx** [reply]
> > We don't use sdl for audio.
> 
> okay. So what do you use guys? Cause all the time just in one moment sound is gone and I need to change audio device to other and back sometimes in the game, sometimes in default settings of the system. It's annoying. Fedora 43, pipewire, Plasma 6.5

The audio library used is [BASS](https://www.un4seen.com/), which only supports ALSA audio output (via `pipewire-alsa`). If you are having issues with that I suggest you make a new discussion with some details about your configuration.

**Reply by djsiropchik** [reply]
> > > We don't use sdl for audio.
> > 
> > 
> > okay. So what do you use guys? Cause all the time just in one moment sound is gone and I need to change audio device to other and back sometimes in the game, sometimes in default settings of the system. It's annoying. Fedora 43, pipewire, Plasma 6.5
> 
> The audio library used is [BASS](https://www.un4seen.com/), which only supports ALSA audio output (via `pipewire-alsa`). If you are having issues with that I suggest you make a new discussion with some details about your configuration.

Sorry it was mistaken. I already found the other thread about sound. But what we gonna do with the video. It will not gonna work until this issue will not be fixed? https://github.com/ValveSoftware/steam-for-linux/issues/8020

**Comment by Redapple-one** [comment]
osu runs in wayland will increase input latency even with ``SDL_VIDEODRIVER=wayland``
see [#37338](https://github.com/ppy/osu/discussions/37338)
